docs/configuration.md

```diff
@@ -0,0 +1,696 @@
+
+* `nonAvvmBalances` in _genesis data_ depends on `testBalance`.
+  Let's describe how balances and addresses generated in this case:
+
+  - To generated balances we first find limit on how much
+  coins we can generate by subtracting sum of all real avvm balances
+  (considering `avvmBalanceFactor`) from maximal possible coin value
+  (45000000000000000). Total generated value is the minimum between
+  `testBalance.totalBalance` and the aforementioned limit.
+
+    Let's say
+    that `n` is the difference between total generated value and the sum
+    of all fake avvm balances (`count * oneBalance`). Then we have
+    `richmenBalance = richmenShare * n`. There will be
+    `testBalance.richmen` addresses with approximately `richmenBalance /
+    testBalance.richmen` coins. The rest will be evenly distributed
+    among `testBalance.poors` addresses.
+
+  - For each richman we generate a simple PubKey addresses without
+    attributes.
+
+    Poor addresses are either HD addresses with empty
+    passphrase or simple PubKey addresses depending on
+    `useHDAddresses` value. All addresses have BootstrapEra stake
+    distribution.
+
+    Note: rich addresses are derived from secret keys of
+    richmen which may be different from bootstrap stakeholders'
+    keys. It depends on `useHeavyDlg`.
+
+#### System start time
+
+With _genesis spec_, system start time is taken from command line option
+(`--system-start`). The
+reason why system start is not part
+of `initializer` is that dev clusters are to be launched
+often, and it's easier to change this value from CLI flags rather then configuration file.
+
+### Tools
+
+There are some tools relevant to genesis data.
+
+* `cardano-keygen` has a command to dump all genesis secrets to
+  files. Usage: `cardano-keygen --system-start 0 --configuration-file
+  <file> --configuration-key <key> --configuration-seed <seed>
+  generate-keys-by-spec --genesis-out-dir <dir>`. This command will
+  generate and dump secrets to `<dir>`. To deploy a cluster you need
+  keys of core nodes. They can be found in
+  `<dir>/generated-keys/rich/`. It can be used only when genesis spec
+  is used.
+
+* `cardano-keygen` also has command to dump genesis data (in JSON
+  format). Usage: `cardano-keygen --system-start SYSTEM_START
+  --configuration-file <file> --configuration-key <key>
+  --configuration-seed <seed> dump-genesis-data --path
+  <file>`. Default path is `genesis-data.json`. It can be used to
+  generate _genesis data_ from _genesis spec_ if you want it to be
+  used by nodes or just want to verify generated data.
+
+* `cardano-node-simple` and `cardano-node` also have command line option to
+  dump genesis data. The option is
+  `--dump-genesis-data-to genesis.json` (data will be dumped to
+  `genesis.json`).
+
+### Generating genesis for testnet
+
+For testnet we can't use `spec` for genesis, at least because
+converting genesis spec to genesis data is currently very slow and we
```

Comment:
It's not really obvious how these statements are related (you use "and"). I suppose that hd address generation is a bottleneck?